pulp_rpm/app/migrations/0040_DATA_change_subrepo_naming_scheme.py

```diff
@@ -0,0 +1,24 @@
+from django.db import migrations
+
+
+def change_subrepo_names(apps, schema_editor):
+    RpmRepository = apps.get_model('rpm', 'RpmRepository')
+
+    repos = list()
+    for repo in RpmRepository.objects.filter(user_hidden=True).only("name").iterator():
+        if repo.name.count("-") == 1:
+            repo.name = "-".join(repo.name, str(repo.pk))
```

Comment:
Also FWIW I'm pretty sure you were right the first time, we have to 

> go through every dist tree and check which repos its addons and variants refer to.

so I need to go back and do that part.

Reply:
Yeah, unfortunately. Reading code now, I see that a repository_pk of a main repo is used, not the subrepo, that means we need to go through every dist tree.
It's not too late to change approach and support multiple naming schemes :D